R/half.life.R

```diff
@@ -118,11 +126,18 @@ pk.calc.half.life <- function(conc, time, tmax, tlast,
   # if (inherits(data$conc, "units")) {
   #   conc_units <- units(data$conc)
   # } else {
-    conc_units <- NULL
+  conc_units <- NULL
   #}
   data$log_conc <- log(data$conc)
-  # as.numeric() to handle units objects
+  # Filter out points with 0 concentration. as.numeric() to handle units objects
   data <- data[as.numeric(data$conc) > 0,]
+  if (!is.null(time.dose)) {
+    # If multiple doses are within the interval, consider only the last one
+    time.dose = time.dose[length(time.dose)]
+    duration.dose = duration.dose[length(duration.dose)]
+    if (!is.na(time.dose) && !is.na(duration.dose))
+      data <- data[as.numeric(data$time) > as.numeric(time.dose) + as.numeric(duration.dose), ]

```

Comment:
Please use `max(time.dose + duration.dose, na.rm = TRUE)` for the time to filter after. There can be many unusual scenarios with mixing time and durations of doses and the first dose to start is not always the first one to end (e.g. multiple boluses during a long-duration infusion of pain medication).